backend/src/layer/data/flickr_api/test/mod.rs

```diff
@@ -0,0 +1,2 @@
+#[cfg(test)]
```

Comment:
if you want to make the entire test module compile only on tests (what you should), you should use `#![cfg(test)]` which then targets the entire module (file).
But i think adding this annotation to the `mod test` in the parent module is clearer.